src/Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -1,186 +1,226 @@
+            Logger.Log("Running " + cswinrtExe + " " + arguments);
+
+            var processInfo = new ProcessStartInfo
+            {
+                FileName = cswinrtExe,
+                Arguments = arguments,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                WindowStyle = ProcessWindowStyle.Hidden,
+                CreateNoWindow = true
+            };
+
+            try
+            {
+                using var cswinrtProcess = Process.Start(processInfo);
+                Logger.Log(cswinrtProcess.StandardOutput.ReadToEnd());
+                Logger.Log(cswinrtProcess.StandardError.ReadToEnd());
+                cswinrtProcess.WaitForExit();
+
+                if (cswinrtProcess.ExitCode != 0)
+                {
+                    throw new Win32Exception(cswinrtProcess.ExitCode);
+                }
+
+                foreach (var file in Directory.GetFiles(outputDir, "*.cs", SearchOption.TopDirectoryOnly))
+                {
+                    Logger.Log("Adding " + file);
+                    context.AddSource(Path.GetFileNameWithoutExtension(file), SourceText.From(File.ReadAllText(file), Encoding.UTF8));
+                }
+            }
+            finally
+            {
+                if (!GetKeepGeneratedSources(context))
+                {
+                    Directory.Delete(outputDir, true);
+                }
+            }
+        }
+
+        private string GetWinmdOutputFile(GeneratorExecutionContext context)
+        {
+            return Path.Combine(GetGeneratedFilesDir(context), GetAssemblyName(context) + ".winmd");
+        }
+
+        private void GenerateWinMD(MetadataBuilder metadataBuilder, string outputFile)
+        {
+            Logger.Log("Writing " + outputFile);
+            var managedPeBuilder = new ManagedPEBuilder(
+                new PEHeaderBuilder(
+                    machine: Machine.I386,
+                    imageCharacteristics: Characteristics.ExecutableImage | Characteristics.Dll | Characteristics.Bit32Machine),
+                new MetadataRootBuilder(metadataBuilder, "WindowsRuntime 1.4"),
+                new BlobBuilder(),
+                flags: CorFlags.ILOnly);
+
+            var peBlob = new BlobBuilder();
+            managedPeBuilder.Serialize(peBlob);
+
+            using var fs = new FileStream(outputFile, FileMode.Create, FileAccess.Write);
+            peBlob.WriteContentTo(fs);
+        }
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            if (!IsCsWinRTComponent(context))
+            {
+                return;
+            }
+
+            Logger.Initialize(context);
+
+            try
+            {
+                context.AddSource("System.Runtime.InteropServices.WindowsRuntime", SourceText.From(@"

```

Comment:
Have you tried this out in a test app? It has no problem "resolving" the attribute? Just curious

Reply:
The test app builds fine, but intellisense does complain about it.  I might end up removing it from here and moving it to winrt.runtime later, but I wanted to avoid adding another DLL that we need to worry about.  I am hoping that the intellisense issue goes away with VS / source generator improvements.